object.c

```diff
@@ -2349,9 +2360,33 @@ rb_mod_attr_reader(int argc, VALUE *argv, VALUE klass)
     VALUE names = rb_ary_new2(argc);
 
     for (i=0; i<argc; i++) {
-        ID id = id_for_attr(klass, argv[i]);
-        rb_attr(klass, id, TRUE, FALSE, TRUE);
-        rb_ary_push(names, ID2SYM(id));
+        VALUE name_sym = argv[i];
+        VALUE name_str = rb_sym2str(name_sym);
```

Comment:
`Module#attr_reader` (and so on) accepts attribute name strings as well as symbols, while `rb_sym2str` argument must be `T_SYMBOL`.

Reply:
Ah, right, that makes perfect sense. Thanks for explaining!